src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +74,651 @@ public ByteArraySet createSmallSet() {
+    }
+
+    @Test
+    public void reallyBigRandomSerializeDerializeSizeCheckRemove() {
+        testRandomSerializeDerializeSizeCheckRemove(SIZE_REALLY_BIG);
+    }
+
+    private void testSimilarRemove(boolean forceMerger) {
+        System.out.println("Testing remove method for similar inputs with " + forceMerger + " forceMerger");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = new ArrayList<>();
+        vals.add(new ByteArray("bytear"));
+        vals.add(new ByteArray("ByteAr"));
+        vals.add(new ByteArray("bytea0"));
+        vals.add(new ByteArray("0ytear"));
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        assertEquals(4, set.size());
+        Collections.shuffle(vals);
+
+        if (forceMerger) {
+            set.serialize();
+        }
+
+        List<ByteArray> deletedVals = new ArrayList<>();
+        for(ByteArray val: vals) {
+            assertTrue(set.remove(val));
+            assertFalse(set.contains(val));
+
+            deletedVals.add(val);
+            assertEquals(4, set.size() + deletedVals.size());
+        }
+    }
+
+    @Test
+    public void testMergedSimilarRemove() {
+        testSimilarRemove(true);
+    }
+
+    @Test
+    public void testUnmergedSimilarRemove() {
+        testSimilarRemove(false);
+    }
+
+    private void testToArray(int size) {
+        System.out.println("Testing toArray method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        if (size > 0) {
+            assertFalse(set.isEmpty());
+        } else {
+            assertTrue(set.isEmpty());
+        }
+
+        List<ByteArray> alreadySeenVals = new ArrayList<>();
+        int seenCount = 0;
+        for(ByteArray val: set.toArray()) {
+            assertFalse(alreadySeenVals.contains(val));
+            alreadySeenVals.add(val);
+            assertTrue(vals.contains(val));
+            seenCount++;
+        }
+        assertEquals(size, seenCount);
+    }
+
+    @Test
+    public void emptyToArray() {
+        testToArray(SIZE_EMPTY);
+    }
+
+    @Test
+    public void reallySmallToArray() {
+        testToArray(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallToArray() {
+        testToArray(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularToArray() {
+        testToArray(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigToArray() {
+        testToArray(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigToArray() {
+        testToArray(SIZE_REALLY_BIG);
+    }
+
+    private void testContains(int size) {
+        System.out.println("Testing contains method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
```

Comment:
Can be instantiated by the constructor rather than looping

Reply:
ByteArraySet cannot be instantiated with a List<ByteArray>, but simplified with addAll